app/src/pages/admin/allUsersPage/allUsersGrid/nameColumn/nameColumn.jsx

```diff
@@ -14,23 +18,82 @@ const messages = defineMessages({
   youLabel: { id: 'NameColumn.youLabel', defaultMessage: 'you' },
   adminLabel: { id: 'NameColumn.adminLabel', defaultMessage: 'admin' },
   makeAdminLabel: { id: 'NameColumn.makeAdminLabel', defaultMessage: 'make admin' },
+  changeAccountRoleText: {
+    id: 'NameColumn.changeAccountRoleText',
+    defaultMessage: "Are you sure you want to change the account role for the '{name}' ?",
+  },
+  changeAccountRoleTitle: {
+    id: 'NameColumn.changeAccountRoleTitle',
+    defaultMessage: 'Change role',
+  },
+  submitText: { id: 'NameColumn.submitText', defaultMessage: 'Change' },
+  cancelText: { id: 'NameColumn.cancelText', defaultMessage: 'Cancel' },
+  changeAccountRoleNotification: {
+    id: 'NameColumn.changeAccountRoleNotification',
+    defaultMessage: "User role for '{name}' was changed.",
+  },
 });
 
-@connect((state) => ({
-  currentUser: userIdSelector(state),
-}))
+@connect(
+  (state) => ({
+    currentUser: userIdSelector(state),
+  }),
+  {
+    showModal: showModalAction,
+    fetchAllUsers: fetchAllUsersAction,
+    showNotification,
+  },
+)
 @injectIntl
 export class NameColumn extends Component {
   static propTypes = {
     className: PropTypes.string.isRequired,
     value: PropTypes.object,
     currentUser: PropTypes.string,
     intl: intlShape.isRequired,
+    showModal: PropTypes.func,
+    changeAccountRole: PropTypes.func,
+    fetchAllUsers: PropTypes.func,
+    showNotification: PropTypes.func.isRequired,
   };
   static defaultProps = {
     value: {},
     currentUser: '',
+    showModal: () => {},
+    changeAccountRole: () => {},
+    fetchAllUsers: () => {},
   };
+
+  onChangeAccountRole = () => {
+    const { intl, showModal, value, fetchAllUsers } = this.props;
+    const onSubmit = () => {
+      fetch(URLS.userInfo(value.userId), {
+        method: 'PUT',
+        data: {
+          role: value.userRole === ADMINISTRATOR ? USER : ADMINISTRATOR,
+        },
+      }).then(fetchAllUsers);
+      this.props.showNotification({
+        type: NOTIFICATION_TYPES.SUCCESS,
+        message: intl.formatMessage(messages.changeAccountRoleNotification, {
+          name: value.fullName,
+        }),
+      });
+    };
+
+    showModal({
+      id: 'allUsersChangeProjectRoleModal',
+      data: {
+        message: intl.formatMessage(messages.changeAccountRoleText, { name: value.fullName }),
```

Comment:
These localization messages can be a part of this modal. No need to make them as parameters